internal/impl/sql/integration_test.go

```diff
@@ -1546,3 +1547,120 @@ func waitForRds(localstackPort string) (err error) {
 	}
 	return errors.New("polling localstack for ready RDS failed")
 }
+
+func TestIntegrationXxx(t *testing.T) {
```

Comment:
```suggestion
func TestIntegrationReconnectToDatabase(t *testing.T) {
```

Perhaps we should add an additional test-case to the integration testing suite for the case of an upstream disconnection 👀